packages/plugin-core/src/_extension.ts

```diff
@@ -796,25 +813,51 @@ async function showWelcomeOrWhatsNew({
         previousExtensionVersion,
       });
       await StateService.instance().setGlobalVersion(version);
+
+      // ^t6dxodie048o
+      const toastOrView = UPGRADE_TOAST_OR_VIEW_TEST.getUserGroup(
+        SegmentClient.instance().anonymousId
+      );
+
       AnalyticsUtils.track(VSCodeEvents.Upgrade, {
         previousVersion: previousExtensionVersion,
         duration: getDurationMilliseconds(start),
+        toastOrView,
       });
-      vscode.window
-        .showInformationMessage(
-          `Dendron has been upgraded to ${version} from ${previousExtensionVersion}`,
-          "See what changed"
-        )
-        .then((resp) => {
-          if (resp === "See what changed") {
-            vscode.commands.executeCommand(
-              "vscode.open",
-              vscode.Uri.parse(
-                "https://dendron.so/notes/9bc92432-a24c-492b-b831-4d5378c1692b.html"
-              )
-            );
-          }
-        });
+
+      switch (toastOrView) {
+        case UpgradeToastOrViewTestEnum.upgradeToast: {
+          vscode.window
+            .showInformationMessage(
+              `Dendron has been upgraded to ${version} from ${previousExtensionVersion}`,
+              "See what changed"
+            )
+            .then((resp) => {
+              if (resp === "See what changed") {
+                AnalyticsUtils.track(
+                  VSCodeEvents.UpgradeSeeWhatsChangedClicked,
+                  {
+                    previousVersion: previousExtensionVersion,
+                    duration: getDurationMilliseconds(start),
+                  }
+                );
+                vscode.commands.executeCommand(
+                  "vscode.open",
+                  vscode.Uri.parse(
+                    "https://dendron.so/notes/9bc92432-a24c-492b-b831-4d5378c1692b.html"
+                  )
+                );
+              }
+            });
+          break;
+        }
+        case UpgradeToastOrViewTestEnum.upgradeView: {
+          showUpgradeView();
```

Comment:
can we track a "success action" for the upgrade view? for the toast, success is if they clicked the button to see whats changed. what is success when showing the view? (eg. did user scroll on the iframe or click a link)

Reply:
I don't think we can track links clicked since we have the page loaded inside a webview (I think the cross-origin policy blocks that). Scrolling is also not interesting because the changelog in most weeks fits within the screen, so the user can read everything without scrolling.

I added a time measurement instead which measures how long the webview was visible for. I added an event for the view closing, and attached the duration to that. We can decide when we're looking at the telemetry how long of an interaction means the view was successful.

Reply:
@kevinslin - yeah I think understanding 'success' for this particular feature will be slightly difficult.  I like Kaan's approach here, with the caveat that the webview will stay open if someone has Dendron upgraded, but then walks away from their computer.

Another thing we can look at is if the A/B group starts interacting with the new features outlined in the changelog in different ways (i.e. are they more likely to use features from the new week, etc.)